src/chia_log/log_consumer.py

```diff
@@ -50,49 +54,56 @@ def _notify_subscribers(self, logs: str):
 
 
 class FileLogConsumer(LogConsumer):
-    """Specific implementation for a simple file consumer"""
-
     def __init__(self, log_path: Path):
-        logging.info("Enabled file log consumer.")
         super().__init__()
-        self._log_path = log_path
+        self._expanded_log_path = str(log_path.expanduser())
+        self._offset_path = PurePath("debug.log.offset")
         self._is_running = True
         self._thread = Thread(target=self._consume_loop)
         self._thread.start()
+        self._log_size = 0
 
     def stop(self):
         logging.info("Stopping")
         self._is_running = False
 
+    @retry((FileNotFoundError, PermissionError), delay=2)
     def _consume_loop(self):
-        expanded_user_log_path = str(self._log_path.expanduser())
-        logging.info(f"Consuming log file from {expanded_user_log_path}")
+        while self._is_running:
+            for log_line in Pygtail(self._expanded_log_path, read_from_end=True, offset_file=self._offset_path):
+                self._notify_subscribers(log_line)
 
-        if is_win_platform():
-            consume_command_args = ["powershell.exe", "get-content", expanded_user_log_path, "-tail", "1", "-wait"]
-        else:
-            consume_command_args = ["tail", "-F", expanded_user_log_path]
 
-        f = subprocess.Popen(consume_command_args, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        while self._is_running:
-            log_line = f.stdout.readline().decode(encoding="utf-8")
-            self._notify_subscribers(log_line)
+class PosixFileLogConsumer(FileLogConsumer):
+    """Specific implementation for a simple file consumer for Linux/MacOS"""
+
+    def __init__(self, log_path: Path):
+        logging.info("Enabled Posix file log consumer.")
+        super(PosixFileLogConsumer, self).__init__(log_path)
+
+
+class WindowsFileLogConsumer(FileLogConsumer):
+    """Specific implementation for a simple file consumer for Windows"""
+
+    def __init__(self, log_path: Path):
+        logging.info("Enabled Windows file log consumer.")
+        super(WindowsFileLogConsumer, self).__init__(log_path)
```

Comment:
It just occurred to me that we don't anymore need separate log consumers for PosixFile and WindowsFile anymore?

Reply:
Done! Initially left it in primarily because of consistency with NetworkConsumer, but you're right; it adds unwanted complexity